Fix a typo in many FP8 tests like TestFP8MatmulCPU.test_scaled_mm_block_wise_numerics_bfloat16_lhs_block_128_rhs_block_1_M_256_N_768_K_512_test_case_data_random_calc_scales_cpu
> ValueError: Invalid scaling configuration.
Fix extracted from https://github.com/pytorch/pytorch/pull/176064

It also doesn't work on CPU at all with the same error, see https://github.com/pytorch/pytorch/pull/176064
Hence only run on CUDA.

author: Alexander Grund (TU Dresden)
diff --git a/test/test_scaled_matmul_cuda.py b/test/test_scaled_matmul_cuda.py
@@ -1361,6 +1361,7 @@ class TestFP8Matmul(TestCase):
         else:
             test()
 
+    @onlyCUDA
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8 or IS_WINDOWS, f8_msg)
     @unittest.skipIf(not IS_SM90, "DeepSeek style (1x128, 128x128) blockwise scaling requires SM90 (Hopper)")
     @unittest.skipIf(
--- a/test/test_scaled_matmul_cuda.py
+++ b/test/test_scaled_matmul_cuda.py
@@ -1543,13 +1543,13 @@ class TestFP8Matmul(TestCase):
             y = torch.eye(M, device=device)
 
             x_hp, x_recipe, x_fp8, x_scales, x_scales_original = _build_lhs(x, lhs_block)
-            y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_lhs(y, rhs_block)
+            y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_rhs(y, rhs_block)
         elif test_case == "x_ones_y_ones_calc_scales":
             x = torch.full((M, K), 1.0, device=device)
             y = torch.full((N, K), 1.0, device=device)
 
             x_hp, x_recipe, x_fp8, x_scales, x_scales_original = _build_lhs(x, lhs_block)
-            y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_lhs(y, rhs_block)
+            y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_rhs(y, rhs_block)
         elif test_case in ["x_ones_y_ones_set_scales", "x_ones_y_ones_modify_scales"]:
             x = torch.full((M, K), 1.0, device=device)
             y = torch.full((N, K), 1.0, device=device)
@@ -1584,7 +1584,7 @@ class TestFP8Matmul(TestCase):
             y = torch.randn(N, K, device=device, dtype=output_dtype) * 1e-3
 
             x_hp, x_recipe, x_fp8, x_scales, x_scales_original = _build_lhs(x, lhs_block)
-            y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_lhs(y, rhs_block)
+            y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_rhs(y, rhs_block)
         else:
             raise ValueError("Unknown test-case passed")
 
